src/engine/bufferscalers/enginebufferscalerubberband.cpp

```diff
@@ -98,6 +101,9 @@ void EngineBufferScaleRubberBand::setScaleParameters(double base_rate,
 }
 
 void EngineBufferScaleRubberBand::onSampleRateChanged() {
+    // TODO: Resetting the sample rate will cause internal
+    // memory allocations that may block the real-time thread.
+    // When is this function actually invoked??
```

Comment:
If the sample rate is being changed, the audio is restarting anyway.

Reply:
I just had to adapt this sparsely documented code that may have unknown side-effects. I didn't have the intention to change the behaviour. I left this comment for the next one who stumbles upon it. That's all I can do for now to address the review comments.

Reply:
okay